src/PerformanceTests/Common/Scenarios/SagaRetrievalRunner.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+
+using NServiceBus;
+using Tests.Permutations;
+using Variables;
+
+partial class SagaRetrievalRunner : BaseRunner, IThrowIfPermutationIsNotAllowed
+{
+    public void ThrowIfPermutationIsNotAllowed(Permutation permutation)
+    {
+        if (permutation.Persister == Persistence.InMemory)
+            throw new InvalidOperationException("This test has no use running with InMemory persistence.");
+    }
+
+    protected override void Start()
+    {
+    }
+
+    protected override void Stop()
+    {
+    }
+
+    public class Command : ICommand
+    {
+        public Command(int id)
```

Comment:
Did you test this with the serializers? i would not assume that all serializers can deserialize a message with constructor that has args.